pkg/webhook/immutable_fields_validator.go

```diff
@@ -320,6 +332,45 @@ func validateImmutableFieldsForTFBasedResource(obj, oldObj *unstructured.Unstruc
 	return allowedResponse
 }
 
+func validateImmutableFieldsForLoggingLogMetricResource(oldSpec, spec map[string]interface{}) admission.Response {
+	if isResourceIDModified(oldSpec, spec) {
+		return admission.Errored(http.StatusForbidden,
+			k8s.NewImmutableFieldsMutationError([]string{k8s.ResourceIDFieldPath}))
+	}
+
+	isImmutableFieldsModified := func(oldSpec, newSpec map[string]interface{}, field string) bool {
+		tokens := strings.Split(field, ".")
+		oldVal, ok1, err1 := unstructured.NestedFieldCopy(oldSpec, tokens...)
+		newVal, ok2, err2 := unstructured.NestedFieldCopy(newSpec, tokens...)
+		if oldVal == nil && newVal == nil {
+			return false
+		}
+		if !ok1 || err1 != nil {
+			return true
+		}
+		if !ok2 || err2 != nil {
+			return true
+		}
+		return !reflect.DeepEqual(oldVal, newVal)
+	}
+	ImmutableFields := []string{}
```

Comment:
Nit: `immutableFields` because it isn't a global, and also probably simplest to do:

```
immutableFields := []string{
  "foo",
  "bar",
}
```